test/rulesets/Base/mapreduce.jl

```diff
@@ -203,4 +267,60 @@ end
             @test unthunk(back(fill(0.5, 2, 2))[2]) ≈ [1/2 0; 0 0]  # ProjectTo'd to Diagonal now
         end
     end
+    @testset "accumulate(f, ::Array)" begin
+        CFG = ChainRulesTestUtils.ADviaRuleConfig()
+
+        # Simple
+        y1, b1 = rrule(CFG, accumulate, *, [1,2,3,4]; init=1)
+        @test y1 == [1, 2, 6, 24]
+        @test b1([1,1,1,1]) == (NoTangent(), NoTangent(), [33, 16, 10, 6])
+
+        y2, b2 = rrule(CFG, accumulate, /, [1 2; 3 4])
+        @test y2 ≈ accumulate(/, [1 2; 3 4])
+        @test b2(ones(2,2))[3] ≈ [1.5416666 -0.104166664; -0.18055555 -0.010416667]  atol=1e-6
+        # ForwardDiff.gradient(x -> sum(accumulate(/,x)), Float32[1 2; 3 4]) |> string
+
+        # Test execution order
+        c3 = Counter()
+        y3, b3 = rrule(CFG, accumulate, c3, [5,7,11]; init=3)
+        @test c3 == Counter(3)
+        @test y3 == [8, 30, 123] == accumulate(Counter(), [5,7,11]; init=3)
+        @test b3([1,1,1]) == (NoTangent(), NoTangent(), [29169, 602, 23]) # the 23 is clear!
+
+        c4 = Counter()
+        y4, b4 = rrule(CFG, accumulate, c4, [5,7,11])
+        @test c4 == Counter(2)
+        @test y4 == [5, (5+7)*1, ((5+7)*1 + 11)*2] == accumulate(Counter(), [5,7,11])
+        @test b4([1,1,1]) == (NoTangent(), NoTangent(), [417, 42*(1 + 12), 22])
+
+        # Test gradient of function
+        y7, b7 = rrule(CFG, accumulate, Multiplier(3), [5,7,11])
+        @test y7 == accumulate((x,y)->x*y*3, [5,7,11])
+        @test b7([1,1,1]) == (NoTangent(), Tangent{Multiplier{Int}}(x = 2345,), [715, 510, 315])
+        # ForwardDiff.derivative(z -> sum(accumulate((x,y)->x*y*z, [5,7,11])), 3) == 2345
+        # ForwardDiff.gradient(z -> sum(accumulate((x,y)->x*y*3, z)), [5,7,11]) |> string
+
+        y8, b8 = rrule(CFG, accumulate, Multiplier(13), [5,7,11], init=3)
+        @test y8 == [195, 17745, 2537535] == accumulate((x,y)->x*y*13, [5,7,11], init=3)
+        @test b8([1,1,1]) == (NoTangent(), Tangent{Multiplier{Int}}(x = 588330,), [511095, 365040, 230685])
+        # ForwardDiff.derivative(z -> sum(accumulate((x,y)->x*y*z, [5,7,11], init=3)), 13)
+        # ForwardDiff.gradient(z -> sum(accumulate((x,y)->x*y*13, z, init=3)), [5,7,11]) |> string
```

Comment:
Don't forget to remove these

Reply:
My vote is to leave these, or something like them, as explanation of where 511095 comes from. The simpler Counter ones you can do by hand, which is why I wrote `(5+7)*1` etc.